### PR TITLE
Allow add custom shared dict

### PR DESCRIPTION
**Pull request by griffenliu**
NOTE: Please read the Contributing.md guidelines before submitting your patch:

https://github.com/apache/incubator-apisix/blob/master/Contributing.md#how-to-add-a-new-feature-or-change-an-existing-one

### Summary

Allows you to add custom shared dict through configuration

### Full changelog

* config.yaml : lua_shared_dicts
* apisix

### CODE REVIEW

**Comment by moonming** [comment]
If you need to add one shared dict, then you will using this shared dict in Lua code, right?
So is this PR works with custom plugin?

**Comment by griffenliu** [comment]
> If you need to add one shared dict, then you will using this shared dict in Lua code, right?
> So is this PR works with custom plugin?

Yes, I need add some custom plugin. I used the mlcache library, which needs to add some shared caches.

**Comment by moonming** [comment]
Got it.

File: conf/config.yaml [hunk 1]
```diff
@@ -111,6 +111,8 @@ nginx_config:                     # config for render the template to genarate n
     real_ip_from:                  # http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
       - 127.0.0.1
       - 'unix:'
+    lua_shared_dicts:
+      ipc_shared_dict: 100m         # custom shared cache
```
Comment (moonming) [inline conf/config.yaml]: should be commented by default.

File: conf/config.yaml [hunk 2]
```diff
@@ -111,8 +111,8 @@ nginx_config:                     # config for render the template to genarate n
     real_ip_from:                  # http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
       - 127.0.0.1
       - 'unix:'
-    lua_shared_dicts:
-      ipc_shared_dict: 100m         # custom shared cache
+    lua_shared_dicts:              # add custom shared cache to nginx.conf
+      ipc_shared_dict: 100m        # custom shared cache, format: `cache-key: cache-size`
```
Comment (moonming) [inline conf/config.yaml]: should be `# ipc_shared_dict: 100m`, which is only a demo.

Comment (griffenliu) [reply conf/config.yaml]: ok, thanks~

**Pull request merged**